src/DataTypes/Serializations/SerializationString.cpp

```diff
@@ -235,7 +235,7 @@ void SerializationString::deserializeBinaryBulk(IColumn & column, ReadBuffer & i
     {
         try
         {
-            data.reserve(size_to_reserve);
```

Comment:
I'm also not sure about using of `reserve_exact` in serializations because while reading `deserializeBinaryBulk` is called for each granule and that will mean `realloc` for each granule which may be too much.